ctk/common/src/test/kotlin/org/codice/compilance/verification/profile/subject/confirmations/HolderOfKeySubjectConfirmationVerificationSpec.kt

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compilance.verification.profile.subject.confirmations
+
+import io.kotlintest.forAll
+import io.kotlintest.matchers.string.shouldContain
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.StringSpec
+import org.codice.compliance.Common
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLProfiles_3_1_a
+import org.codice.compliance.SAMLProfiles_3_1_b
+import org.codice.compliance.SAMLProfiles_3_1_c
+import org.codice.compliance.utils.HOLDER_OF_KEY_URI
+import org.codice.compliance.verification.profile.subject.confirmations.HolderOfKeySubjectConfirmationVerification
+import java.time.Instant
+import java.util.UUID
+
+@Suppress("StringLiteralDuplication")
+class HolderOfKeySubjectConfirmationVerificationSpec : StringSpec() {
+    private val correctType = "xsi:type=\"s2:KeyInfoConfirmationDataType\""
+    private val correctKeyInfo = """
+                |<ds:KeyInfo>
+                |  <ds:KeyName>key</ds:KeyName>
+                |</ds:KeyInfo>
+                """.trimMargin()
+
+    init {
+        "response with correct holder-of-key subject confirmation should pass" {
+            Common.buildDom(createResponse(
+                subjConf = createHolderOfKeySubjConf(keyInfo = correctKeyInfo))).let {
+                HolderOfKeySubjectConfirmationVerification(it).verify()
+            }
+        }
+
+        "holder-of-key with no subject confirmation data type should pass" {
+            Common.buildDom(createResponse(
+                subjConf = createHolderOfKeySubjConf("", correctKeyInfo))).let {
+                HolderOfKeySubjectConfirmationVerification(it).verify()
+            }
+        }
```

Comment:
❓ Should we test when the Response has no holderOfKey at all and make sure it passes?